src/StreamJsonRpc.Tests/MessagePackFormatter.cs

```diff
@@ -47,13 +62,28 @@ public MessagePackFormatter(bool compress)
         }
 
         /// <inheritdoc/>
-        public JsonRpcMessage Deserialize(ReadOnlySequence<byte> contentBuffer)
+        public JsonRpc Rpc
         {
-            return this.compress
-                ? (JsonRpcMessage)LZ4MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream())
-                : (JsonRpcMessage)MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream());
+            private get => this.rpc;
+            set
+            {
+                Verify.Operation(this.rpc == null, "This formatter already belongs to another JsonRpc instance. Create a new instance of this formatter for each new JsonRpc instance.");
+
+                this.rpc = value;
+            }
         }
 
+        /// <summary>
+        /// Gets <see cref="MessagePackSerializerOptions.LZ4Default"/> if <see cref="compress"/> is true, otherwise <c>null</c>.
+        /// </summary>
+        private MessagePackSerializerOptions CompressOptions
```

Comment:
>CompressOptions [](start = 45, length = 15)

nit: rename to `SerializerOptions` since there's more to it than just whether it's compressed. Right *now* that's all the thought we put into it, but to the user of this property, they just use it as the serializer options.